redwood-protocol-guest/src/commonMain/kotlin/app/cash/redwood/protocol/guest/ProtocolRedwoodComposition.kt

```diff
@@ -48,21 +49,28 @@ public fun ProtocolRedwoodComposition(
     saveableStateRegistry = saveableStateRegistry,
     uiConfigurations = uiConfigurations,
     widgetSystem = bridge.widgetSystem,
+  )
+  return ProtocolRedwoodComposition(
+    composition,
+    widgetVersion
   ) {
-    bridge.getChangesOrNull()?.let(changesSink::sendChanges)
+    key(bridge.compositionsCount) {
+      bridge.getChangesOrNull()?.let(changesSink::sendChanges)
+    }
```

Comment:
I needed to defer this side effect until after the 'late changes' callback happened, otherwise we'd get two batches of changes with the remove in the first, and the add in the second, and the add would fail because the element wouldn’t exist on the host anymore.

Reply:
Unfortunately, it’s still broken with a similar root cause . . . the host app crashes because we get an add() that precedes the corresponding remove():

```
FATAL EXCEPTION: main
Process: com.example.redwood.emojisearch.android.views, PID: 11706
java.lang.IllegalStateException: The specified child already has a parent. You must call removeView() on the child's parent first.
	at android.view.ViewGroup.addViewInner(ViewGroup.java:5247)
	at android.view.ViewGroup.addView(ViewGroup.java:5076)
	at android.view.ViewGroup.addView(ViewGroup.java:5016)
	at android.view.ViewGroup.addView(ViewGroup.java:4988)
	at app.cash.redwood.lazylayout.view.ViewLazyList$ViewHolder.setContent(ViewLazyList.kt:286)
	at app.cash.redwood.lazylayout.view.ViewLazyList$processor$1.setContent(ViewLazyList.kt:92)
	at app.cash.redwood.lazylayout.view.ViewLazyList$processor$1.setContent(ViewLazyList.kt:74)
	at app.cash.redwood.lazylayout.widget.LazyListUpdateProcessor$Binding.bind$redwood_lazylayout_widget(LazyListUpdateProcessor.kt:402)
	at app.cash.redwood.lazylayout.widget.LazyListUpdateProcessor.bind(LazyListUpdateProcessor.kt:302)
	at app.cash.redwood.lazylayout.view.ViewLazyList$LazyContentItemListAdapter.onBindViewHolder(ViewLazyList.kt:263)
	at app.cash.redwood.lazylayout.view.ViewLazyList$LazyContentItemListAdapter.onBindViewHolder(ViewLazyList.kt:233)
```

Reply:
I think this change moves the callback to fire earlier, not later. The old one fired on `Applier.onEndChanges` which is after recomposition has walked the whole function tree. You are now triggering it within the function tree.

Reply:
Yeah, I need a `SideEffect {}` here, which should run after all the composition stuff.